src/lib.rs

```diff
@@ -140,14 +157,18 @@ impl Lottery {
             let win_player_index = *self.players.keys().nth(index).expect("Player not found");
             let player = self.players[&win_player_index];
 
-            if self.token_address.is_some() {
+            if let Some(token_address) = self.token_address {
```

Comment:
We might borrow `token_address` instead of copying it if we use a reference later.
```suggestion
            if let Some(ref token_address) = self.token_address {
```